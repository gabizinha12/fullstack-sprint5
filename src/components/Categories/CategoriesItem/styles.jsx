import styled from "styled-components";

export const ListItem = styled.li``;

export const ItemLink = styled.a`
  font-size: 0.8125rem;
  font-weight: 600;
  color: #000;
  margin-right: 10px;
  display: flex;
  text-transform: uppercase;
  padding-bottom: 5px;
  vertical-align: baseline;
  &:hover {
    border-bottom: 2px solid ${({ theme }) => theme.text};
  }
`;

export const LinkSpan = styled.span``;
