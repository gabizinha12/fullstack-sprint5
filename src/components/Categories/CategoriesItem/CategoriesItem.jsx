import { ListItem, ItemLink, LinkSpan } from "./styles";

export function CategoriesItem({ label, link }) {
  return (
    <ListItem>
      <ItemLink href={link}>
        <LinkSpan>{label}</LinkSpan>
      </ItemLink>
    </ListItem>
  );
}
