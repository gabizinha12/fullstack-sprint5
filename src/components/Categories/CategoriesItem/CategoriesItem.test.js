import { render, screen } from "@testing-library/react";
import { CategoriesItem } from "./CategoriesItem";

describe("Teste de snapshot CategoriesItem", () => {
  it("should render CategoriesItem", () => {
    const { container } = render(<CategoriesItem link="teste" label="abc" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
