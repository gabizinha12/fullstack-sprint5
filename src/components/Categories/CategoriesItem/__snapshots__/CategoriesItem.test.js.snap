// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de snapshot CategoriesItem should render CategoriesItem 1`] = `
<li
  class="sc-bdnxRM"
>
  <a
    class="sc-gtsrHT bjlWZU"
    href="teste"
  >
    <span
      class="sc-dlnjwi"
    >
      abc
    </span>
  </a>
</li>
`;
